### PR TITLE
Fix the names of two sets of weight and bias in mcore_to_nemo_mapping of scripts/convert_gpt_nemo_to_mcore.py

### PR DESCRIPTION
**Pull request by renweizhukov**
These two sets of weight and bias have been renamed according to the `module_name_rewrite_list` given in https://github.com/NVIDIA/Megatron-LM/blob/e33c8f78a35765d5aa37475a144da60e8a2349d1/megatron/core/inference/gpt/state_dict_hooks.py#L116-L119

Also change the underscores in the two command-line options to hyphens to follow the convention of using hyphens in command-line options. Note that `argparse` automatically converts hyphens in argument names to underscores in the attributes of the namespace returned by `parse_args()`.

# What does this PR do ?

Update the names of the two sets of weight and bias according to latest Megatron-LM.

**Collection**: NLP

# Changelog 
- Update the names of the two sets of weight and bias according to latest Megatron-LM.
- Change the underscores in the two command-line options to hyphens to follow the convention of using hyphens in command-line options.

# Usage

https://docs.nvidia.com/nemo-framework/user-guide/latest/modelalignment/dpo.html

```
python convert_gpt_nemo_to_mcore.py --input-name-or-path ./model_checkpoint --output-path ./mcore_gpt.nemo
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [x] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests? The model conversion script does not have any unit test.
- [x] Did you add or update any necessary documentation? Update the comment in the script.
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [x] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to #9626

### CODE REVIEW

File: scripts/checkpoint_converters/convert_gpt_nemo_to_mcore.py [hunk 1]
```diff
@@ -40,22 +40,22 @@
...
-        "--input_name_or_path",
+        "--input-name-or-path",
         type=str,
         default=None,
         required=True,
         help="Path to extracted, TP1 PP1 NeMo GPT checkpoint.",
     )
     parser.add_argument(
-        "--output_path",
+        "--output-path",
```
Comment (yaoyu-33) [inline scripts/checkpoint_converters/convert_gpt_nemo_to_mcore.py:58]: can you don't change this to dash, keep it as underscore

File: scripts/checkpoint_converters/convert_gpt_nemo_to_mcore.py [hunk 2]
```diff
@@ -40,22 +40,22 @@
...
+ --input-name-or-path <path to extracted, TP1 PP1 legacy checkpoint folder> \
+ --output-path <path to output nemo file>
 """
 
 
 def get_args():
     parser = ArgumentParser()
     parser.add_argument(
-        "--input_name_or_path",
+        "--input-name-or-path",
```
Comment (yaoyu-33) [inline scripts/checkpoint_converters/convert_gpt_nemo_to_mcore.py:51]: can you don't change this to dash, keep it as underscore

Comment (renweizhukov) [reply scripts/checkpoint_converters/convert_gpt_nemo_to_mcore.py:51]: I changed the underscores to dashes to be consistent with the two other options `--cpu-only` and `--ignore-if-missing`. Besides, in command-line interfaces, the convention is to use hyphens (-) rather than underscores (_) to separate words in option names. It aligns with the usage in Unix-style commands, where hyphens are used as option prefixes.

File: scripts/checkpoint_converters/convert_gpt_nemo_to_mcore.py (hunk 1 above)
Comment (renweizhukov) [reply scripts/checkpoint_converters/convert_gpt_nemo_to_mcore.py:58]: See my response in https://github.com/NVIDIA/NeMo/pull/9628#discussion_r1691806797

File: scripts/checkpoint_converters/convert_gpt_nemo_to_mcore.py (hunk 2 above)
Comment (yaoyu-33) [reply scripts/checkpoint_converters/convert_gpt_nemo_to_mcore.py:51]: hi we should probably change others then. We try to keep the core args the same (using _) across the checkpoint_converters folder.

**Review by yaoyu-33 — APPROVED** [review]
